packages/jaeger-ui/src/components/TracePage/ArchiveNotifier/index.tsx

```diff
@@ -93,32 +89,24 @@ function updateNotification(oldState: ENotifiedState | null, nextState: ENotifie
   }
 }
 
-function processProps(notifiedState: ENotifiedState | null, props: Props) {
-  const nxNotifiedState = getNextNotifiedState(props);
-  updateNotification(notifiedState, nxNotifiedState, props);
-  return nxNotifiedState;
-}
+const ArchiveNotifier: React.FC<Props> = props => {
+  const notifiedStateRef = useRef<ENotifiedState | null>(null);
 
-export default class ArchiveNotifier extends React.PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    const notifiedState = processProps(null, props);
-    this.state = { notifiedState };
-  }
+  useEffect(() => {
+    const nextNotifiedState = getNextNotifiedState(props);
+    updateNotification(notifiedStateRef.current, nextNotifiedState, props);
+    notifiedStateRef.current = nextNotifiedState;
+  }, [props.archivedState]);
 
-  static getDerivedStateFromProps(props: Props, state: State) {
-    const notifiedState = processProps(state.notifiedState, props);
-    return { notifiedState };
-  }
+  useEffect(() => {
```

Comment:
so this function will always be executed (since it has no dependencies listed)? Previously it was only called when component was unmounting.